### PR TITLE
Logout in oc10 while click logout in web 

### PR DESCRIPTION
**Pull request by JanAckermann**
## Related Issue

- Fixes https://github.com/owncloud/web/issues/5886

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-runtime/src/store/user.js [hunk 1]
```diff
@@ -34,17 +34,23 @@ const actions = {
...
-    const logoutFinalizer = (forceRedirect = false) => {
-      // Remove signed in user
-      dispatch('cleanUpLoginState')
-      dispatch('hideModal')
-      dispatch('loadSettingsValues')
+  async logout(context) {
+    const logoutFinalizer = (isOauth2 = false) => {
+      // Force redirect
+      if (isOauth2) {
+        const logoutUrl = context.getters?.configuration?.auth?.logoutUrl
```
Comment (kulmann) [inline packages/web-runtime/src/store/user.js:41]: Would it make sense to use configuration.server concatenated with "index.php/logout" as fallback?

File: packages/web-runtime/src/store/user.js [hunk 2]
```diff
@@ -34,17 +34,23 @@ const actions = {
...
+        if (logoutUrl) {
+          return (window.location = logoutUrl)
+        }
 
-      // Force redirect to login
-      if (forceRedirect) {
         router.push({ name: 'login' })
       }
+
+      // Remove signed in user
```
Comment (kulmann) [inline packages/web-runtime/src/store/user.js:50]: Don't we need the store cleanups before the redirect?

File: packages/web-runtime/src/store/user.js (hunk 1 above)
Comment (JanAckermann) [reply packages/web-runtime/src/store/user.js:41]: good idea!

File: packages/web-runtime/src/store/user.js (hunk 2 above)
Comment (JanAckermann) [reply packages/web-runtime/src/store/user.js:50]: Well, this is something I can't answer yet, we do a redirect, so I thought the store should be cleaned while accessing the page the next time. 
I moved this code path, as there is already a redirect happening in one of these dispatched actions

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/25502/67/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-runtime/src/store/user.js [hunk 3]
```diff
@@ -31,18 +31,26 @@ const actions = {
     // clear dynamic navItems
     context.dispatch('clearDynamicNavItems')
 
-    // clear oidc client state
-    vueAuthInstance.clearLoginState()
+    if (options.clearOIDCLoginState) {
+      // clear oidc client state
+      vueAuthInstance.clearLoginState()
+    }
```
Comment (JanAckermann) [inline packages/web-runtime/src/store/user.js:37]: @kulmann this code path triggers a redirect already and interferes with redirecting to the oc10 log out,
I don't call ` vueAuthInstance.clearLoginState()`anymore if we are logged in via oauth2, is this ok?

Comment (kulmann) [reply packages/web-runtime/src/store/user.js:37]: Is it really sufficient for oauth2 to skip that step? o_O

**Review by kulmann — APPROVED** [review]
LGTM 🚀 
FYI, found some other issues with cyclic action dispatching... will tackle that in a subsequent PR that refactors most of the auth related code. 😅

**Comment by pascuflow** [comment]
Pulled latest from master to test this, after logging out of Web, if you then log in with different credentials, you are still logged in as previous user.

**Comment by pascuflow** [comment]
Fixed by adding `sessionStorage.clear()` before logout() is called in web/packages/web-runtime/src/components/Topbar/UserMenu.vue

**Pull request merged**